src/main/flight/pid.c

```diff
@@ -1458,9 +1458,13 @@ void FAST_CODE pidController(const pidProfile_t *pidProfile, timeUs_t currentTim
         }
     }
 
+    // When PASSTHRU_MODE is active - reset all PIDs to zero so the aircraft won't snap out of control 
+    // because of accumulated PIDs once PASSTHRU_MODE gets disabled.
+    float isFixedWingAndPassthru = isFixedWing() && FLIGHT_MODE(PASSTHRU_MODE);
+
     // Disable PID control if at zero throttle or if gyro overflow detected
     // This may look very innefficient, but it is done on purpose to always show real CPU usage as in flight
-    if (!pidRuntime.pidStabilisationEnabled || gyroOverflowDetected()) {
+    if (!pidRuntime.pidStabilisationEnabled || gyroOverflowDetected() || isFixedWingAndPassthru) {
```

Comment:
Why resetting all PID outputs `pidResetIterm()` is not enough?

Reply:
this way it would be clearly visible in BB logs that PID controller is not doing anything

Reply:
> this way it would be clearly visible in BB logs that PID controller is fully disabled

(flight modes are recorded in bb. But keep it if you think it is better this way)

> Also both P and D terms may in fact have kind of longterm accumulated effect

doesn't it work like PID 'warmup' - starting with steady state of filters?

Reply:
oh wait, are you suggesting to zero out gyroRateDterm for example?

Reply:
tbh I don't think it could be a problem either way